smartapps/smartthings/ecobee-connect.src/ecobee-connect.groovy

```diff
@@ -494,6 +534,21 @@ def initialize() {
 	state.initializeEndTime = now()
 }
 
+def purgeUninstalledDeviceData() {
+	// purge state from devices that are not selected
+	def thermostatList = state.thermostats ?: [:]
+	def remoteSensors = state.remoteSensors2 ?: [:]
+	def switchList = state.switchList ?: [:]
+
+	// clean up device lists
+	thermostatList.keySet().removeAll(thermostatList.keySet() - thermostats)
+	remoteSensors.keySet().removeAll(remoteSensors.keySet() - ecobeesensors)
+	switchList.keySet().removeAll(switchList.keySet() - ecobeeswitches)
+	state.thermostats = thermostatList
```

Comment:
nit: Can we just do `state.thermostats = thermostatList.keySet().removeAll(thermostatList.keySet() - thermostats)` over here?

Reply:
Maybe, if you trust the state assignment, nothing I've tested though